### PR TITLE
Remove global spike detector and related functions/variables

### PR DESCRIPTION
**Pull request by jakobj**
As discussed in the last NEST meeting, this PR is working towards removal of the global spike detector. I've removed all obvous parts of the code, but there might be some other parts still hiding, so I need the help of competent reviewers, @mschmidt87 @suku248, please have a look at this and comment.

### CODE REVIEW

File: nestkernel/mpi_manager.cpp [hunk 1]
```diff
@@ -130,7 +128,7 @@ nest::MPIManager::init_mpi( int* argc, char** argv[] )
 void
 nest::MPIManager::initialize()
 {
-  set_num_rec_processes( 0, true );
+  kernel().rng_manager.create_rngs_();
```
Comment (suku248) [inline nestkernel/mpi_manager.cpp]: Is this really required? `RNGManager::initialize()` also calls `create_rngs_()`

File: nestkernel/vp_manager_impl.h [hunk 2]
```diff
@@ -49,54 +49,20 @@ VPManager::is_local_vp( thread vp ) const
...
-    % ( kernel().mpi_manager.get_num_sim_processes() * get_num_threads() );
-}
-
-inline thread
-VPManager::suggest_rec_vp( index gid ) const
-{
-  return gid
-    % ( kernel().mpi_manager.get_num_rec_processes() * get_num_threads() )
-    + kernel().mpi_manager.get_num_sim_processes() * get_num_threads();
+  return gid % ( kernel().mpi_manager.get_num_processes() * get_num_threads() );
```
Comment (suku248) [inline nestkernel/vp_manager_impl.h]: Use `kernel().vp_manager.get_num_virtual_processes()` here

File: nestkernel/rng_manager.h [hunk 3]
```diff
@@ -44,7 +44,7 @@ namespace nest
 
 class RNGManager : public ManagerInterface
 {
-  friend void MPIManager::set_num_rec_processes( int, bool ); // create_rngs_
+  friend void MPIManager::initialize(); // calls create_rngs_
```
Comment (suku248) [inline nestkernel/rng_manager.h]: Might not be required. See [previous comment](https://github.com/nest/nest-simulator/pull/653/files?diff=unified#r101513246)

File: models/spike_detector.h [hunk 4]
```diff
@@ -103,18 +103,12 @@ class spike_detector : public Node
...
   }
-  bool
-  potential_global_receiver() const
-  {
-    return true;
-  }
-  void set_local_receiver( const bool lr );
+
   bool
   local_receiver() const
```
Comment (jakobj) [inline models/spike_detector.h]: I think this is needed, see, e.g., https://github.com/nest/nest-simulator/blob/master/nestkernel/connection_manager.cpp#L380

Comment (suku248) [inline models/spike_detector.h]: You are right. And it should return `true` -- no member` local_receiver_` required as it can no longer be set.

**Comment by jakobj** [comment]
thanks @mschmidt87 and @suku248 for the reviews. i have addressed your comments now. please have another look.

**Comment by jougs** [comment]
@jakobj: not yet. I'm hoping to get the merges finished by Sunday.

**Comment by jougs** [comment]
The changes have been merged to my [remove-sibling-container](https://github.com/jougs/nest-simulator/tree/remove-sibling-container) branch and will enter master via another PR. I'm thus closing this one. Thanks for the work!

**Pull request closed without merging**